src/lib/InstantSearch.js

```diff
@@ -24,6 +24,8 @@ function defaultCreateURL() { return '#'; }
  * @param  {function} [options.searchFunction] A hook that will be called each time a search needs to be done, with the
  * helper as a parameter. It's your responsibility to call helper.search(). This option allows you to avoid doing
  * searches at page load for example.
+ * @param   {function} [options.clientFactory] A function called upon initialization with the `options.appId`
```

Comment:
Because it's a verb-based name ("createXX") option, if I read the name of this option without its doc, I'd assume it's a boolean to say whether or not the client should be created.

I prefer a noun, which informs on the nature of the option.
Would you prefer `algoliaClientFactory`?